### PR TITLE
Search: Modularize Customizer and DOM integration

### PR DESCRIPTION
**Pull request by jsnmoon**
#### Changes proposed in this Pull Request:
In preparation for using our Instant Search application within Gutenberg, this PR modularizes `<SearchApp>` to better isolate its Customizer and DOM integration.

#### Jetpack product discussion
Necessary for our future customization efforts, see p1HpG7-c6N-p2.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* Apply this change to your Jetpack site with a Search subscription.
* Ensure that the search interface works as expected.

### CODE REVIEW

File: projects/plugins/jetpack/modules/search/instant-search/components/dom-event-handler.jsx
Comment (kangzj) [inline projects/plugins/jetpack/modules/search/instant-search/components/dom-event-handler.jsx:152]: The line triggers the following error in console and overlay wouldn't popup.
undefinded-overlay-options

`overlayOptions` doesn't seem to be defined as a state, probably need to be passed as a props from search-app.

**Review by kangzj — CHANGES_REQUESTED** [review]
The refactoring decouples cusomizer and DOM integration, so much clearer and will it easier for future integrations. Just need to have a look at the undefined error commented. Great work Jason👍

File: projects/plugins/jetpack/modules/search/instant-search/components/dom-event-handler.jsx
Comment (jsnmoon) [reply projects/plugins/jetpack/modules/search/instant-search/components/dom-event-handler.jsx]: Good catch! I discovered and fixed several more scope issues with DomEventHandler while fixing this.

**Comment by jsnmoon** [comment]
Ready for another round of reviews!

**Review by kangzj — DISMISSED** [review]
Looks and works well. Well done! 👍

**Review by jeherve — DISMISSED** [review]
This tests well, although I suppose we may want to wait until #20023 is merged and rebase?

**Comment by sdixon194** [comment]
Merged https://github.com/Automattic/jetpack/pull/20023, so this can be rebased.

**Comment by jsnmoon** [comment]
Rebased and ready for another round of reviews!

**Review by jeherve — APPROVED** [review]

**Pull request merged**
